app/javascript/controllers/work_type_controller.js

```diff
@@ -23,17 +24,59 @@ export default class extends Controller {
     this.areaTarget.hidden = false
   }
 
+  toggleMoreTypes(event) {
```

Comment:
Most of the changes below are for twiddling/hiding/unhiding links and sections of the page as the user interacts with the different widgets. See the screencast in the PR description to see it work.